sdk/storage/azure-storage-file-datalake/azure/storage/file/datalake/_data_lake_directory_client.py

```diff
@@ -0,0 +1,512 @@
+        :keyword str if_match:
+            An ETag value, or the wildcard character (*). Specify this header to perform
+            the operation only if the resource's ETag matches the value specified.
+        :keyword str if_none_match:
+            An ETag value, or the wildcard character (*). Specify this header
+            to perform the operation only if the resource's ETag does not match
+            the value specified. Specify the wildcard character (*) to perform
+            the operation only if the resource does not exist, and fail the
+            operation if it does exist.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :return: DataLakeFileClient
+        """
+        file_client = self.get_file_client(file)
+        file_client.create_file(**kwargs)
+        return file_client
+
+    def get_file_client(self, file  # type: Union[FileProperties, str]
+                        ):
+        # type: (...) -> DataLakeFileClient
+        """Get a client to interact with the specified file.
+
+        The file need not already exist.
+
+        :param file:
+            The file with which to interact. This can either be the name of the file,
+            or an instance of FileProperties. eg. directory/subdirectory/file
+        :type file: str or ~azure.storage.file.datalake.FileProperties
+        :returns: A DataLakeFileClient.
+        :rtype: ~azure.storage.file.datalake..DataLakeFileClient
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/test_datalake_service_samples.py
+                :start-after: [START bsc_get_file_client]
+                :end-before: [END bsc_get_file_client]
+                :language: python
+                :dedent: 12
+                :caption: Getting the file client to interact with a specific file.
+        """
+        try:
+            file_path = file.name
+        except AttributeError:
+            file_path = self.path_name + '/' + file
+
+        return DataLakeFileClient(
+            self.url, self.file_system_name, file_path, credential=self._raw_credential,
+            _hosts=self._hosts, _configuration=self._config, _pipeline=self._pipeline,
+            _location_mode=self._location_mode, require_encryption=self.require_encryption,
+            key_encryption_key=self.key_encryption_key,
+            key_resolver_function=self.key_resolver_function)
+
+    def get_sub_directory_client(self, sub_directory  # type: Union[DirectoryProperties, str]
+                                 ):
+        # type: (...) -> DataLakeDirectoryClient
+        """Get a client to interact with the specified subdirectory of the current directory.
+
+        The sub subdirectory need not already exist.
+
+        :param sub_directory:
+            The directory with which to interact. This can either be the name of the directory,
+            or an instance of DirectoryProperties.
```

Comment:
Same here, looks like the parameter only takes the name of the directory

Reply:
Thanks for catching this! I forgot to change 😀